### PR TITLE
new how-to for google maps itinerary link

### PR DESCRIPTION
**Pull request by itsakifa**
add an automatic field in the attribute form containing a link to open google maps and navigate to or show the feature.

### CODE REVIEW

File: documentation/how-to/itinerary.en.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+## Configure attribute form widget in QGIS
+:material-monitor: Desktop preparation
+
+Here is an example for navigation to features of a point layer.
+
+Create a new field in your data table (type text). In the attribute form settings, select "attachment" as widget type. Tick "Display a hyperlink for document path (read-only)".
+Then enter the following expression as default value:
+
+*concat( 'https://www.google.com/maps/dir/?api=1&destination=', y(transform($geometry, layer_property(@layer, 'crs'),'EPSG:4326')), '%2C', x(transform($geometry, layer_property(@layer, 'crs'),'EPSG:4326')), '&travelmode=driving')*
```
Comment (suricactus) [inline documentation/how-to/itinerary.en.md:19]:
````suggestion
```
concat(
  'https://www.google.com/maps/dir/?api=1&destination=',
  y(transform($geometry, layer_property(@layer, 'crs'), 'EPSG:4326')),
  '%2C',
  x(transform($geometry, layer_property(@layer, 'crs'), 'EPSG:4326')),
  '&travelmode=driving'
)
```
````

I think newlines vastly improve readability.

File: documentation/how-to/itinerary.en.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+And tick "Apply default value on update" in case you make changes to your geometry.
+
+If you simply want to show your feature location in Google Maps, you can use the following expression:
+*concat( 'https://maps.google.com?q=  ',y(transform($geometry, layer_property(@layer, 'crs'),'EPSG:4326')), '%2C', x(transform($geometry, layer_property(@layer, 'crs'),'EPSG:4326')), '&zoom=19&t=h')*
+
+## Usage
+:material-tablet: Fieldwork
+
+Click on the feature on the map where you want navigation to or that you want to open in Google Maps. In the attribute form, click on the link towards Google Maps.
+
```
Comment (m-kuhn) [inline documentation/how-to/itinerary.en.md:30]:
```suggestion
```

This will make the pre-commit check happy

**Comment by suricactus** [comment]
@itsakifa @m-kuhn shall we merge if green? Looks like a nice piece of docs.

File: documentation/how-to/itinerary.en.md [hunk 3]
```diff
@@ -16,7 +16,15 @@ Here is an example for navigation to features of a point layer.
 Create a new field in your data table (type text). In the attribute form settings, select "attachment" as widget type. Tick "Display a hyperlink for document path (read-only)".
 Then enter the following expression as default value:
 
-*concat( 'https://www.google.com/maps/dir/?api=1&destination=', y(transform($geometry, layer_property(@layer, 'crs'),'EPSG:4326')), '%2C', x(transform($geometry, layer_property(@layer, 'crs'),'EPSG:4326')), '&travelmode=driving')*
+```
```
Comment (m-kuhn) [inline documentation/how-to/itinerary.en.md:19]:
```suggestion
```sql
```

For the extra-fancyness of syntax highlighting

Comment (suricactus) [reply documentation/how-to/itinerary.en.md]: QField Docs - Fancy!

**Comment by itsakifa** [comment]
@suricactus if you think that it's different enough from this https://docs.qfield.org/how-to/attributes-form/#define-qml-widgets? I've seen in the meantime that a similar description exists - even if it's not exactly the same

**Review by suricactus — APPROVED** [review]
I think it is nice and worth it.

**Pull request merged**